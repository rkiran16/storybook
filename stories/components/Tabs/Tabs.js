import { html, nothing } from "lit-html";

function onWindowLoad() {
    window.addEventListener('load', (event) => {
        const tabs = document.querySelectorAll(".nav-link");
        if (tabs) {
            tabs.forEach((ele) => {
                ele.addEventListener("click", handleClick);
            })
        }
    });
}

function handleClick(event) {
    event.target.scrollIntoView({
        block: 'nearest',
        inline: 'center',
        behavior: 'smooth'
    })
}

const TabIcon = () => html`<span class="material-icons ms-3">shopping_cart</span>`;

const TabItem = (tabName, index, disabled, hasIcon, label="Label") => html`
    <li class="nav-item" role="presentation">
        <button 
            class=${`nav-link d-flex align-items-center ${index === 0 ? 'active' : '' }`} 
            id=${`Tab-${index}`} 
            type="button" 
            role="tab" 
            data-bs-toggle="tab" 
            data-bs-target=${`#Tab-${index}-pane`} 
            aria-controls=${`Tab-${index}-pane`} 
            aria-selected="true"
            ?disabled=${disabled} 
        >
            ${`${tabName} ${label} ${index+1}`}
            ${hasIcon ? TabIcon() : nothing} 
        </button>
    </li>`

const TabContent = (tabName, index) => html`
    <div 
        class=${`tab-pane py-2 ps-4 fade show ${index === 0 ? 'active' : '' }`} 
        id=${`Tab-${index}-pane`} 
        role="tabpanel" 
        aria-labelledby=${`Tab-${index}`}
    >
        <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
            Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
            Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
            Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
    </div>
`

export const Tabs = ({disabled = false, tabCount = 4, alignment, hasIcon = false, whiteActiveTab = false}) => {
    const whiteBackgroundClass = whiteActiveTab ? 'white-active-tab' : '';
    return html`
        <script>${onWindowLoad()}</script>
        <ul class=${`nav nav-tabs p-2 ${alignment} ${whiteBackgroundClass}`} id="myTab" role="tablist">
            ${Array(tabCount).fill("Tab").map((ele, index) => TabItem(ele, index, disabled, hasIcon))}
        </ul>
        <div class="tab-content" id="myTabContent">
            ${Array(tabCount).fill("Tab").map((ele, index) => TabContent(ele, index))}
        </div>
    `;
}