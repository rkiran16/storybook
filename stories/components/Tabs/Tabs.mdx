import { Canvas, Meta, Story } from "@storybook/addon-docs";

# Tabs

<p>
    The tabs component is a flexible navigational utility. It can accommodate
    any number of tabs and will scroll upon overflow. Alignment options are left,
    center and right.
</p>

<p>
    Please note that scrolling will not work properly in the <code>center</code> alignment
    due to part of the child container overflowing past the start of the parent container and part
    past the end. Because of this, it is necessary for the tabs to align to the left or right at a
    breakpoint prior to needing to scroll.
</p>

<p>
    Javascript is necessary to automate centering the clicked tab if overflow is occurring. This
    will need to be implemented for each instance of the tab, and can be done so with the following
    javascript function:
</p>

<p>
    <code>scrollIntoView(&#123;block: 'nearest', inline: 'center', behavior: 'smooth'&#125;);</code>
</p>

## Options

<code>Show Icon</code>
<p>Allows for an svg icon on each tab</p>

<code>Tab Alignment</code>
<p>
    Allows for the alignment of tabs to the left, right, or center of the parent container. Note
    that if center alignment is used, the tabs will need to align to one side BEFORE the need for
    scrolling occurs as the screen size moves towards tablet/mobile.
</p>

<code>White background for active tab</code>
<p>Optional setting to make the background of the active tab white, for use on non-white backgrounds</p>


<Canvas>
  <Story id="components-tabs--standard" name="Default Modal"></Story>
</Canvas>