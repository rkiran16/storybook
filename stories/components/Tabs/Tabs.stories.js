import { Tabs } from "./Tabs";
import docs from "./Tabs.mdx";

export default {
   title: "Components/Tabs",
   parameters: {
    zeplinLink: "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/62b4a1b7f3c68b1ed9a3e4ab",
    docs: {
      page: docs
    }
   },
   argTypes: {
    tabCount: {
      name: "Number of tabs",
      control: {type: "select"},
      defaultValue: 4,
      options:[1,2,3,4,7,10]
    },
    disabled: {
      name: "Disable tabs",
      type: { name: 'boolean', required: false},
      description: "Use when the user cannot proceed or until an input is collected",
      defaultValue: false,
      table: {
        type: { summary: 'boolean' },
        defaultValue: { summary: false },
      },
      control: {
        type: 'boolean'
      }
    },
    hasIcon: {
      name: "Show icon",
      type: { name: 'boolean', required: false},
      defaultValue: false,
      table: {
        type: { summary: 'boolean' },
        defaultValue: { summary: false },
      },
      control: {
        type: 'boolean'
      }
    },
    alignment: {
      name: "Tab alignment",
      control: { type: 'select'},
      description: "Change the horizontal alignment of your nav with flexbox utilities.By default, navs are left-aligned, but you can easily change them to center or right aligned.",
      defaultValue: "left",
      options: ['left', 'center', 'right'],
      mapping: {
        left: 'left-align',
        center: 'center-align',
        right: 'right-align'
      },
    },
    whiteActiveTab: {
      name: "White background for active tab",
      control: { type: 'boolean', required: false},
      defaultValue: false,
      table: {
        type: { summary: 'boolean' },
        defaultValue: { summary: false },
      },
      control: { 
        type: 'boolean'
      }
    }
  }
}

const Template = args => Tabs(args);

export const Standard = Template.bind({});
Standard.args = {
  disabled: false
}

export const Disabled = Template.bind({})
Disabled.args = {
   disabled:true
}