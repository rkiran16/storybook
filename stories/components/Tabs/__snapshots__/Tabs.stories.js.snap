// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Tabs Disabled 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
        <script>",
    "</script>
        <ul class=",
    " id=\\"myTab\\" role=\\"tablist\\">
            ",
    "
        </ul>
        <div class=\\"tab-content\\" id=\\"myTabContent\\">
            ",
    "
        </div>
    ",
  ],
  "values": Array [
    undefined,
    "nav nav-tabs p-2 left-align ",
    Array [
      Object {
        "_$litType$": 1,
        "strings": Array [
          "
    <li class=\\"nav-item\\" role=\\"presentation\\">
        <button 
            class=",
          " 
            id=",
          " 
            type=\\"button\\" 
            role=\\"tab\\" 
            data-bs-toggle=\\"tab\\" 
            data-bs-target=",
          " 
            aria-controls=",
          " 
            aria-selected=\\"true\\"
            ?disabled=",
          " 
        >
            ",
          "
            ",
          " 
        </button>
    </li>",
        ],
        "values": Array [
          "nav-link d-flex align-items-center active",
          "Tab-0",
          "#Tab-0-pane",
          "Tab-0-pane",
          true,
          "Tab Label 1",
          Symbol(lit-nothing),
        ],
      },
      Object {
        "_$litType$": 1,
        "strings": Array [
          "
    <li class=\\"nav-item\\" role=\\"presentation\\">
        <button 
            class=",
          " 
            id=",
          " 
            type=\\"button\\" 
            role=\\"tab\\" 
            data-bs-toggle=\\"tab\\" 
            data-bs-target=",
          " 
            aria-controls=",
          " 
            aria-selected=\\"true\\"
            ?disabled=",
          " 
        >
            ",
          "
            ",
          " 
        </button>
    </li>",
        ],
        "values": Array [
          "nav-link d-flex align-items-center ",
          "Tab-1",
          "#Tab-1-pane",
          "Tab-1-pane",
          true,
          "Tab Label 2",
          Symbol(lit-nothing),
        ],
      },
      Object {
        "_$litType$": 1,
        "strings": Array [
          "
    <li class=\\"nav-item\\" role=\\"presentation\\">
        <button 
            class=",
          " 
            id=",
          " 
            type=\\"button\\" 
            role=\\"tab\\" 
            data-bs-toggle=\\"tab\\" 
            data-bs-target=",
          " 
            aria-controls=",
          " 
            aria-selected=\\"true\\"
            ?disabled=",
          " 
        >
            ",
          "
            ",
          " 
        </button>
    </li>",
        ],
        "values": Array [
          "nav-link d-flex align-items-center ",
          "Tab-2",
          "#Tab-2-pane",
          "Tab-2-pane",
          true,
          "Tab Label 3",
          Symbol(lit-nothing),
        ],
      },
      Object {
        "_$litType$": 1,
        "strings": Array [
          "
    <li class=\\"nav-item\\" role=\\"presentation\\">
        <button 
            class=",
          " 
            id=",
          " 
            type=\\"button\\" 
            role=\\"tab\\" 
            data-bs-toggle=\\"tab\\" 
            data-bs-target=",
          " 
            aria-controls=",
          " 
            aria-selected=\\"true\\"
            ?disabled=",
          " 
        >
            ",
          "
            ",
          " 
        </button>
    </li>",
        ],
        "values": Array [
          "nav-link d-flex align-items-center ",
          "Tab-3",
          "#Tab-3-pane",
          "Tab-3-pane",
          true,
          "Tab Label 4",
          Symbol(lit-nothing),
        ],
      },
    ],
    Array [
      Object {
        "_$litType$": 1,
        "strings": Array [
          "
    <div 
        class=",
          " 
        id=",
          " 
        role=\\"tabpanel\\" 
        aria-labelledby=",
          "
    >
        <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
            Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
            Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
            Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
    </div>
",
        ],
        "values": Array [
          "tab-pane py-2 ps-4 fade show active",
          "Tab-0-pane",
          "Tab-0",
        ],
      },
      Object {
        "_$litType$": 1,
        "strings": Array [
          "
    <div 
        class=",
          " 
        id=",
          " 
        role=\\"tabpanel\\" 
        aria-labelledby=",
          "
    >
        <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
            Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
            Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
            Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
    </div>
",
        ],
        "values": Array [
          "tab-pane py-2 ps-4 fade show ",
          "Tab-1-pane",
          "Tab-1",
        ],
      },
      Object {
        "_$litType$": 1,
        "strings": Array [
          "
    <div 
        class=",
          " 
        id=",
          " 
        role=\\"tabpanel\\" 
        aria-labelledby=",
          "
    >
        <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
            Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
            Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
            Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
    </div>
",
        ],
        "values": Array [
          "tab-pane py-2 ps-4 fade show ",
          "Tab-2-pane",
          "Tab-2",
        ],
      },
      Object {
        "_$litType$": 1,
        "strings": Array [
          "
    <div 
        class=",
          " 
        id=",
          " 
        role=\\"tabpanel\\" 
        aria-labelledby=",
          "
    >
        <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
            Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
            Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
            Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
    </div>
",
        ],
        "values": Array [
          "tab-pane py-2 ps-4 fade show ",
          "Tab-3-pane",
          "Tab-3",
        ],
      },
    ],
  ],
}
`;

exports[`Storyshots Components/Tabs Standard 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
        <script>",
    "</script>
        <ul class=",
    " id=\\"myTab\\" role=\\"tablist\\">
            ",
    "
        </ul>
        <div class=\\"tab-content\\" id=\\"myTabContent\\">
            ",
    "
        </div>
    ",
  ],
  "values": Array [
    undefined,
    "nav nav-tabs p-2 left-align ",
    Array [
      Object {
        "_$litType$": 1,
        "strings": Array [
          "
    <li class=\\"nav-item\\" role=\\"presentation\\">
        <button 
            class=",
          " 
            id=",
          " 
            type=\\"button\\" 
            role=\\"tab\\" 
            data-bs-toggle=\\"tab\\" 
            data-bs-target=",
          " 
            aria-controls=",
          " 
            aria-selected=\\"true\\"
            ?disabled=",
          " 
        >
            ",
          "
            ",
          " 
        </button>
    </li>",
        ],
        "values": Array [
          "nav-link d-flex align-items-center active",
          "Tab-0",
          "#Tab-0-pane",
          "Tab-0-pane",
          false,
          "Tab Label 1",
          Symbol(lit-nothing),
        ],
      },
      Object {
        "_$litType$": 1,
        "strings": Array [
          "
    <li class=\\"nav-item\\" role=\\"presentation\\">
        <button 
            class=",
          " 
            id=",
          " 
            type=\\"button\\" 
            role=\\"tab\\" 
            data-bs-toggle=\\"tab\\" 
            data-bs-target=",
          " 
            aria-controls=",
          " 
            aria-selected=\\"true\\"
            ?disabled=",
          " 
        >
            ",
          "
            ",
          " 
        </button>
    </li>",
        ],
        "values": Array [
          "nav-link d-flex align-items-center ",
          "Tab-1",
          "#Tab-1-pane",
          "Tab-1-pane",
          false,
          "Tab Label 2",
          Symbol(lit-nothing),
        ],
      },
      Object {
        "_$litType$": 1,
        "strings": Array [
          "
    <li class=\\"nav-item\\" role=\\"presentation\\">
        <button 
            class=",
          " 
            id=",
          " 
            type=\\"button\\" 
            role=\\"tab\\" 
            data-bs-toggle=\\"tab\\" 
            data-bs-target=",
          " 
            aria-controls=",
          " 
            aria-selected=\\"true\\"
            ?disabled=",
          " 
        >
            ",
          "
            ",
          " 
        </button>
    </li>",
        ],
        "values": Array [
          "nav-link d-flex align-items-center ",
          "Tab-2",
          "#Tab-2-pane",
          "Tab-2-pane",
          false,
          "Tab Label 3",
          Symbol(lit-nothing),
        ],
      },
      Object {
        "_$litType$": 1,
        "strings": Array [
          "
    <li class=\\"nav-item\\" role=\\"presentation\\">
        <button 
            class=",
          " 
            id=",
          " 
            type=\\"button\\" 
            role=\\"tab\\" 
            data-bs-toggle=\\"tab\\" 
            data-bs-target=",
          " 
            aria-controls=",
          " 
            aria-selected=\\"true\\"
            ?disabled=",
          " 
        >
            ",
          "
            ",
          " 
        </button>
    </li>",
        ],
        "values": Array [
          "nav-link d-flex align-items-center ",
          "Tab-3",
          "#Tab-3-pane",
          "Tab-3-pane",
          false,
          "Tab Label 4",
          Symbol(lit-nothing),
        ],
      },
    ],
    Array [
      Object {
        "_$litType$": 1,
        "strings": Array [
          "
    <div 
        class=",
          " 
        id=",
          " 
        role=\\"tabpanel\\" 
        aria-labelledby=",
          "
    >
        <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
            Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
            Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
            Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
    </div>
",
        ],
        "values": Array [
          "tab-pane py-2 ps-4 fade show active",
          "Tab-0-pane",
          "Tab-0",
        ],
      },
      Object {
        "_$litType$": 1,
        "strings": Array [
          "
    <div 
        class=",
          " 
        id=",
          " 
        role=\\"tabpanel\\" 
        aria-labelledby=",
          "
    >
        <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
            Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
            Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
            Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
    </div>
",
        ],
        "values": Array [
          "tab-pane py-2 ps-4 fade show ",
          "Tab-1-pane",
          "Tab-1",
        ],
      },
      Object {
        "_$litType$": 1,
        "strings": Array [
          "
    <div 
        class=",
          " 
        id=",
          " 
        role=\\"tabpanel\\" 
        aria-labelledby=",
          "
    >
        <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
            Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
            Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
            Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
    </div>
",
        ],
        "values": Array [
          "tab-pane py-2 ps-4 fade show ",
          "Tab-2-pane",
          "Tab-2",
        ],
      },
      Object {
        "_$litType$": 1,
        "strings": Array [
          "
    <div 
        class=",
          " 
        id=",
          " 
        role=\\"tabpanel\\" 
        aria-labelledby=",
          "
    >
        <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
            Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
            Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
            Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
    </div>
",
        ],
        "values": Array [
          "tab-pane py-2 ps-4 fade show ",
          "Tab-3-pane",
          "Tab-3",
        ],
      },
    ],
  ],
}
`;
